core-web/libs/template-builder/src/lib/components/template-builder/components/template-builder-box/template-builder-box.component.spec.ts

```diff
@@ -75,4 +107,36 @@ describe('TemplateBuilderBoxComponent', () => {
         expect(paletteButton).toBeTruthy();
         expect(deleteButton).toBeTruthy();
     });
+
+    it('should trigger addContainer when click on plus button', () => {
+        jest.spyOn(spectator.hostComponent, 'addContainer');
+        const addButton = spectator.query(byTestId('btn-plus'));
+
+        addButton.dispatchEvent(new Event('onClick'));
+        expect(spectator.hostComponent.addContainer).toHaveBeenCalled();
+    });
+
+    it('should trigger editStyle when click on palette button', () => {
+        jest.spyOn(spectator.hostComponent, 'editStyle');
+        const paletteButton = spectator.query(byTestId('btn-palette'));
+
+        paletteButton.dispatchEvent(new Event('onClick'));
```

Comment:
I think there is spectator click: spectator.click


` spectator.click(submitButton);`